subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -232,45 +232,55 @@ static void onoff_encode(struct bt_mesh_light_ctrl_srv *srv,
 	}
 }
 
-static int onoff_status_send(struct bt_mesh_light_ctrl_srv *srv,
-			     struct bt_mesh_msg_ctx *ctx,
-			     enum bt_mesh_light_ctrl_srv_state prev_state)
+static int light_onoff_status_send(struct bt_mesh_light_ctrl_srv *srv,
+				   struct bt_mesh_msg_ctx *ctx,
+				   enum bt_mesh_light_ctrl_srv_state prev_state)
 {
 	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_LIGHT_CTRL_OP_LIGHT_ONOFF_STATUS,
 				 3);
-	onoff_encode(srv, &buf, prev_state);
+	light_onoff_encode(srv, &buf, prev_state);
 
 	return model_send(srv->model, ctx, &buf);
 }
 
-static void onoff_pub(struct bt_mesh_light_ctrl_srv *srv,
-		      enum bt_mesh_light_ctrl_srv_state prev_state,
-		      bool pub_gen_onoff)
+static void onoff_encode(struct bt_mesh_light_ctrl_srv *srv,
+			 enum bt_mesh_light_ctrl_srv_state prev_state,
+			 struct bt_mesh_onoff_status *status)
 {
-	onoff_status_send(srv, NULL, prev_state);
+	uint32_t remaining_fade = remaining_fade_time(srv);
```

Comment:
Seems unnecessary to move this out of the else, now it'll just be overwritten afterwards, and same with the target_on_off. Can this be moved back inside the else?